Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2753 @@
+
+- Custom URL Schemes
+- Universal Links
+- UIActivity Sharing
+- App Extensions
+- UIPasteboard
+
+#### Custom URL Schemes
+
+Please refer to the next section "Testing Custom URL Schemes" for more information on what custom URL schemes are and how to test them.
+
+#### Universal Links
+
+##### Overview
+
+Universal links are the iOS equivalent to Android App Links (aka. Digital Asset Links) and are used for deep linking. When a user taps a universal link (to the app's website) he will get seamlessly redirected to the corresponding installed app without going through Safari. If the app isn’t installed, the link will open in Safari.
+
+Universal links are standard web links (HTTP/HTTPS) and are not to be confused with custom URL schemes, which originally were also used for deep linking.
+
+For example, the Telegram app supports both custom URL schemes and universal links:
+
+- `tg://resolve?domain=fridadotre` is a custom URL scheme and uses the `tg://` scheme
+- `https://telegram.me/fridadotre` is a universal link and uses the `https://` scheme
+
+Both result in the same action, the user will be redirected to the specified chat in Telegram ("fridadotre" in this case). However, universal links give several key benefits that are not applicable when using custom URL schemes and are the recommended way to implement deep linking, according to the [Apple Developer Documentation](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html "Universal Links"). Specifically, universal links are:
+
+- **Unique**: Unlike custom URL schemes, universal links can’t be claimed by other apps, because they use standard HTTP or HTTPS links to the app's website. They were introduced as a way to *prevent* URL scheme hijacking attacks (an app installed after the original app may declare the same scheme and the system might target all new requests to the last installed app).
+- **Secure**: When users install the app, iOS downloads and checks a file (`apple-app-site-association`) that was uploaded to the web server to make sure that the website allows the app to open URLs on its behalf. Only the developers can create and upload this file, so the association of their website with the app is secure.
+- **Flexible**: Universal links work even when the app is not installed. Tapping a link to the website would open the content in Safari, as users expect.
+- **Simple**: One URL works for both the website and the app.
+- **Private**: Other apps can communicate with the app without needing to know whether it is installed.
+
+##### Static Analysis
+
+Testing universal links on a static approach includes doing the following:
+
+- Checking the Associated Domains
```

Comment:
Unnecessary capitalization in these points (Maybe not all of them, but definitely in the last item)

Reply:
I wrote it like that because they are the titles of the next subsections (and were originally links to them). Should I remove the capitalization from the list and leave it in the titles?

Example:

```
Testing universal links on a static approach includes doing the following:

- Checking the Associated Domains
- Retrieving the App Site Association File
- Checking the link receiver method
- Checking the data handler method
- Checking if the app is calling other app's universal links

###### Checking the Associated Domains
...
###### Checking if the App is Calling Other App's Universal Links
```

Reply:
If they're no longer links, I would not keep the capitalizations.

Reply:
done